tree/ntuple/v7/inc/ROOT/RDaos.hxx

```diff
@@ -180,19 +181,45 @@ public:
    /// \brief Describes a read/write operation on multiple objects; see the `ReadV`/`WriteV` functions.
    struct RWOperation {
       RWOperation() = default;
-      RWOperation(daos_obj_id_t o, DistributionKey_t d, const std::vector<AttributeKey_t> &as,
-                  const std::vector<d_iov_t> &vs)
-         : fOid(o), fDistributionKey(d), fAttributeKeys(as), fIovs(vs){};
+      RWOperation(daos_obj_id_t o, DistributionKey_t d, std::vector<AttributeKey_t> &&as,
+                  std::vector<std::vector<d_iov_t>> &&vs)
+         : fOid(o), fDistributionKey(d), fAttributeKeys(std::move(as)), fIovVectors(std::move(vs))
+      {
+         for (unsigned i = 0; i < fAttributeKeys.size(); i++)
+            fIndices.emplace(fAttributeKeys[i], i);
+      };
       RWOperation(ROidDkeyPair &k) : fOid(k.oid), fDistributionKey(k.dkey){};
       daos_obj_id_t fOid{};
       DistributionKey_t fDistributionKey{};
       std::vector<AttributeKey_t> fAttributeKeys{};
-      std::vector<d_iov_t> fIovs{};
+      std::vector<std::vector<d_iov_t>> fIovVectors{};
+      std::unordered_map<AttributeKey_t, unsigned> fIndices{};
```

Comment:
This seems somehow partly redundant with `fAttributeKeys`.  I think we might be able to get rid of one of them.

Reply:
The idea behind this was to keep the data vectors directly moveable in `VectorReadWrite` while still accessing the correct IOV vector by map lookup (under the currently-overzealous assumption that attribute keys may show up out of order). 

I committed an update that unifies `fAttributeKeys` and `fIovVectors` into a vector of a custom, generically-named structure `RDaosRequest`, but with insertion still reliant on `fIndices` for now.

Reply:
Insertion is still done IOV-by-IOV (with caging on: page-wise for writing, with attribute key repetition, and cage-wise for reading, with unique `akey`s), so this part has not changed - though it's only beneficial for writing-with-caging for the aforementioned reasons. 
Regardless, I'll have to re-evaluate how IOVs are managed / inserted in the upcoming generic backend refactor this month.